src/query/parser/promql/types.go

```diff
@@ -145,23 +144,29 @@ func newScalarOperator(
 	expr *promql.NumberLiteral,
 	tagOpts models.TagOptions,
 ) (parser.Params, error) {
-	return scalar.NewScalarOp(
-		func(_ time.Time) float64 { return expr.Val },
-		scalar.ScalarType,
-		tagOpts,
-	)
+	return scalar.NewScalarOp(expr.Val, tagOpts)
+}
+
+func isTime(expr promql.Expr) bool {
+	return expr.String() == "time()"
```

Comment:
Hm, surely there's a better way to work out of it's a time expr?  Isn't there a type safe enum like value you can use?

Reply:
Not before it gets parsed unfortunately; alternatively could do some weird stuff where it attempts to edit the parent's params if you receive a `time()` function, but I think that breaks the model in a worse way than this. Figured since parsing is going to be messy and special-casey anyway it won't be a huge difference

Reply:
Ok actually going to refactor it to make it a little less awkward; will rely on block types rather than on prom's param types and remove that janky dependency